test/test_helper.rb

```diff
@@ -103,6 +103,17 @@ def setup_dns_stubs
     Resolv::DNS.any_instance.stubs(:getaddress).raises(Resolv::ResolvError, "DNS must be stub: Resolv::DNS.any_instance.stubs(:getaddress).returns('127.0.0.15')")
     Resolv::DNS.any_instance.stubs(:getaddresses).raises(Resolv::ResolvError, "DNS must be stub: Resolv::DNS.any_instance.stubs(:getaddresses).returns(['127.0.0.15'])")
   end
+
+  def clear_plugins
+    @klass = Foreman::Plugin
+    @plugins_backup = @klass.registered_plugins
+    @registries_backup = @klass.registries
+    @klass.clear
+  end
+
+  def restore_plugins
+    @klass.clear(@plugins_backup, @registries_backup)
+  end
```

Comment:
I wonder if calling `clear_plugins` could set this up automatically as a tear down method. Probably not needed now, but could make it easier to use as a test writer. Or rather, harder to misuse (by forgetting to restore).

Reply:
Done that :+1: